templates/glance/config/00-config.conf

```diff
@@ -33,6 +33,13 @@ use_keystone_limits = {{ .QuotaEnabled }}
 password = {{ .ServicePassword }}
 {{ end }}
 
+[key_manager]
+backend = barbican
+
+[barbican]
+auth_endpoint={{ .KeystoneInternalURL }}
```

Comment:
For reasons I don't understand (but plan to investigate), it seems at some point glance deployments need to generate the service config files when most (all?) of the jinja2 variables have not been defined. Notice all of the "{{ if (index . "SomeVariable") }}" conditionals in this file, including the one at L60.

L40 is not conditional, which leads to this [1] failure.

[1] https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_glance-operator/378/pull-ci-openstack-k8s-operators-glance-operator-main-glance-operator-build-deploy-kuttl/1732439320112926720#1:build-log.txt%3A3592

The short term fix is to add the conditional. The long term question to answer is why all these conditionals are required (cinder doesn't have them).

Reply:
I found out why so many variables in this file are wrapped with a conditional: it's because the top glance controller defines only a small subset (the rest of the variables are defined by the glanceapi controller). Unraveling things to adopt the pattern implemented by the cinder-operator would require non-trivial changes, but I submitted an interim PR https://github.com/openstack-k8s-operators/glance-operator/pull/385 that should help.

Reply:
Yeah, that makes sense and thank you for the quick patch ([PR#385](https://github.com/openstack-k8s-operators/glance-operator/pull/385)).
We really need a minimal `00-config` for `dbsync` and `cronjobs`, so it makes sense to me driving the creation via a single  variable instead of wrapping up every single line of that file.
I'm going to rebase this patch soon and test locally.